debian_package_manager/internal/deb/snapshot.go

```diff
@@ -55,6 +55,25 @@ var (
 func latest(urltemplate string) (string, error) {
 	year, month, _ := time.Now().Date()
 	snapshotURL := fmt.Sprintf(urltemplate, year, month)
+	resp, err := latestFromUrl(snapshotURL)
+	if err != nil {
```

Comment:
I actually think we should only take this path if the currently ignored `day` from time.Now().Date() is "1"

Reply:
> And how should we react if the day is 15? 

So, we work under the assumption that snapshots.debian.org produces snapshot at least daily. We really should only be going to a previous snapshot if there are no snapshot available yet. 

Obtaining indexes could fail for any number of reasons as we generally experience some amount of random failures with this repository. We don't want to accidentally go back a month in that case.